source/material_model/rheology/strain_dependent.cc

```diff
@@ -283,6 +304,29 @@ namespace aspect
         friction_strain_weakening_factors = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Friction strain weakening factors"))),
                                                                                     n_fields,
                                                                                     "Friction strain weakening factors");
+
+        if (prm.get ("Strain healing mechanism") == "no healing")
+          healing_mechanism = no_healing;
+        else if (prm.get ("Strain healing mechanism") == "temperature dependent")
+          {
+            healing_mechanism = temperature_dependent;
+            AssertThrow (this->get_adiabatic_surface_temperature() > 0.0,
```

Comment:
Stylistically, it might be better to move this `AssertThrow` beneath this series of if-else if-else, but I'll let other reviewers comment on that.